web/src/stores/PDFViewerStore.ts

```diff
@@ -1,20 +1,47 @@
-import { defineStore } from 'pinia';
 import { GeneratePdfRequest } from '@/components/documents/models/GeneratePdf';
+import { defineStore } from 'pinia';
 
 export const usePDFViewerStore = defineStore('PDFViewerStore', {
   persist: true,
   state: () => ({
-    documents: [] as GeneratePdfRequest[],
+    storedDocuments: [] as StoreDocument[],
+    groupedDocuments: {} as Record<
+      string,
+      Record<string, StoreDocument[]>
+    >,
   }),
   getters: {
-    documentRequests: (state) => state.documents,
+    documents: (state) => state.storedDocuments,
   },
   actions: {
-    addDocuments(documents: GeneratePdfRequest[]): void {
-      this.documents = [...documents];
+    addDocument(document: StoreDocument): void {
+      this.storedDocuments.push(document);
+
+      if (document.caseNumber) {
+        // Group documents by caseNumber, then by memberName
+        const grouped = this.documents.reduce(
```

Comment:
possible alternative? yours looks fine though:

`const grouped = _.chain(this.documents)
  .groupBy('caseNumber')
  .mapObject((docs) => _.groupBy(docs, 'memberName'))
  .value() as Record<string, Record<string, StoreDocument[]>>;`